File: resources/expected/ros2/fan_in_fan_out/lax/src/fan_in_fan_out_system_cpp_pkg/src/base_code/fanOut_producer_base_src.cpp
#include "fan_in_fan_out_system_cpp_pkg/base_headers/fanOut_producer_base_src.hpp"

//========================================================
// Re-running Codegen will overwrite changes to this file
//========================================================

fanOut_producer_base::fanOut_producer_base() : Node("fanOut_producer")
{
    cb_group_ = this->create_callback_group(rclcpp::CallbackGroupType::Reentrant);
    // Setting up connections
    fanOut_producer_myInteger_publisher_1 = this->create_publisher<fan_in_fan_out_system_cpp_pkg_interfaces::msg::Integer64>(
        "fanOut_consumer1_myInteger",
        1);

    fanOut_producer_myInteger_publisher_2 = this->create_publisher<fan_in_fan_out_system_cpp_pkg_interfaces::msg::Integer64>(
        "fanOut_consumer2_myInteger",
        1);

    // timeTriggered callback timer
    periodTimer_ = this->create_wall_timer(std::chrono::milliseconds(1000),
        std::bind(&fanOut_producer_base::timeTriggered, this), cb_group_);

}

//=================================================
//  C o m m u n i c a t i o n
//=================================================

void fanOut_producer_base::put_myInteger(fan_in_fan_out_system_cpp_pkg_interfaces::msg::Integer64 msg)
{
    fanOut_producer_myInteger_publisher_1->publish(msg);
    fanOut_producer_myInteger_publisher_2->publish(msg);
}

File: resources/expected/ros2/python-fan_in_fan_out/lax/src/fan_in_fan_out_system_py_pkg/fan_in_fan_out_system_py_pkg/base_code/fanOut_producer_base_src.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from queue import Queue
from rclpy.callback_groups import ReentrantCallbackGroup
from fan_in_fan_out_system_py_pkg_interfaces.msg import Integer64

#========================================================
# Re-running Codegen will overwrite changes to this file
#========================================================

class fanOut_producer_base(Node):
    def __init__(self):
        super().__init__("fanOut_producer")

        self.cb_group_ = ReentrantCallbackGroup()

        # Setting up connections
        self.fanOut_producer_myInteger_publisher_1 = self.create_publisher(
            Integer64,
            "fanOut_consumer1_myInteger",
            1)

        self.fanOut_producer_myInteger_publisher_2 = self.create_publisher(
            Integer64,
            "fanOut_consumer2_myInteger",
            1)

        # timeTriggered callback timer
        self.periodTimer_ = self.create_timer(1000, self.timeTriggered, callback_group=self.cb_group_)

    def timeTriggered(self):
        pass

    #=================================================
    #  C o m m u n i c a t i o n
    #=================================================

    def put_myInteger(self, msg):
        typedMsg = Integer64()
        typedMsg.data = msg
        fanOut_producer_myInteger_publisher_1.publish(typedMsg)
        fanOut_producer_myInteger_publisher_2.publish(typedMsg)

